cmd/kcp-catalog/bind/catalogentry/bind.go

```diff
@@ -0,0 +1,212 @@
+	// reference to where CatalogEntry exists. For ex: <absolute_ref_to_workspace>:<catalogEntry>.
+	CatalogEntryRef string
+	// BindWaitTimeout is how long to wait for the apibindings to be created and successful.
+	BindWaitTimeout time.Duration
+}
+
+// NewBindOptions returns new BindOptions.
+func NewBindOptions(streams genericclioptions.IOStreams) *BindOptions {
+	return &BindOptions{
+		Options: base.NewOptions(streams),
+	}
+}
+
+// BindFlags binds fields to cmd's flagset.
+func (b *BindOptions) BindFlags(cmd *cobra.Command) {
+	b.Options.BindFlags(cmd)
+	cmd.Flags().DurationVar(&b.BindWaitTimeout, "timeout", time.Second*30, "Duration to wait for the bindings to be created and bound successfully.")
+}
+
+// Complete ensures all fields are initialized.
+func (b *BindOptions) Complete(args []string) error {
+	if err := b.Options.Complete(); err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		b.CatalogEntryRef = args[0]
+	}
+	return nil
+}
+
+// Validate validates the BindOptions are complete and usable.
+func (b *BindOptions) Validate() error {
+	if b.CatalogEntryRef == "" {
+		return errors.New("`root:ws:catalogentry_object` reference to bind is required as an argument")
+	}
+
+	if !strings.HasPrefix(b.CatalogEntryRef, "root") || !logicalcluster.New(b.CatalogEntryRef).IsValid() {
+		return fmt.Errorf("fully qualified reference to workspace where catalog entry exists is required. The format is `root:<ws>:<catalogentry>`")
+	}
+
+	return b.Options.Validate()
+}
+
+// Run creates an apibinding for the user.
+func (b *BindOptions) Run(ctx context.Context) error {
+	log := klog.FromContext(ctx)
+	config, err := b.ClientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	_, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
+	if err != nil {
+		return err
+	}
+
+	// get the base config, which is needed for creation of clients.
+	baseConfig, err := ctrlcfg.GetConfigWithContext("base")
+	if err != nil {
+		return fmt.Errorf("unable to get base config %v", err)
+	}
+
+	// get the entry referenced in the command to which the user wants to bind.
+	path, entryName := logicalcluster.New(b.CatalogEntryRef).Split()
+	client, err := newCatalogClient(baseConfig, path)
+	if err != nil {
+		return err
+	}
+
+	entry := catalogv1alpha1.CatalogEntry{}
+	err = client.Get(context.TODO(), types.NamespacedName{Name: entryName}, &entry)
+	if err != nil {
+		return fmt.Errorf("cannot find the catalog entry %q referenced in the command in the workspace %q", entryName, path)
+	}
+
+	kcpclient, err := newKCPClusterClient(baseConfig, currentClusterName)
+	if err != nil {
+		return err
+	}
+
+	apiBindings := []apisv1alpha1.APIBinding{}
+	for _, ref := range entry.Spec.Exports {
+		// check if ref is valid. Skip if invalid by logging error.
+		if ref.Workspace.Path == "" || ref.Workspace.ExportName == "" {
+			log.Info("invalid reference %q/%q", ref.Workspace.Path, ref.Workspace.ExportName)
```

Comment:
If you want to report information back to the user, you should use either `b.Out` or `b.ErrOut` with `fmt.Fprint*`